gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2449,16 +2449,33 @@ func evalStaticTypeOf(store Store, last BlockNode, x Expr) Type {
 	t := evalStaticTypeOfRaw(store, last, x)
 	if tt, ok := t.(*tupleType); ok {
 		if len(tt.Elts) != 1 {
-			panic(fmt.Sprintf(
-				"evalStaticTypeOf() only supports *CallExpr with 1 result, got %s",
-				tt.String(),
-			))
-		} else {
-			return tt.Elts[0]
+			loc := last.GetLocation()
+			funcName := getFunctionName(x)
+			valueCount := len(tt.Elts)
+
+			var msg string
+			if valueCount == 0 {
+				msg = fmt.Sprintf("%s: %s() (no value) used as value", loc, funcName)
```

Comment:
Maybe the message could be improved here. "no value" doesn't seem as clear as saying something like "function does not return a value".